src/boot/efi/shim.c

```diff
@@ -97,3 +97,11 @@ EFI_STATUS shim_load_image(EFI_HANDLE parent, const EFI_DEVICE_PATH *device_path
 
         return ret;
 }
+
+void shim_retain_protocol(void) {
+        uint8_t value = 1;
+
+        /* Ask Shim to avoid uninstalling its security protocol, so that we can use it from sd-stub to
+         * validate PE addons. Requires Shim 15.8. */
+        (void) efivar_set_raw(MAKE_GUID_PTR(SHIM_LOCK), u"ShimRetainProtocol", &value, sizeof(value), 0);
```

Comment:
this is a persistent efi var this way. Wich looks really wrong to me. Lacks a EFI_VARIABLE_NON_VOLATILE, no?

Reply:
(yes it's really a stupid name, flag should have been called VARIABLE_PERSISTENT or something)

Reply:
oh, indeed